packages/flutter/lib/src/rendering/object.dart

```diff
@@ -862,15 +857,28 @@ class PipelineOwner {
   ///
   /// See [RendererBinding] for an example of how this function is used.
   void flushLayout() {
+    Map<String, String> debugTimelineArguments = timelineArgumentsIndicatingLandmarkEvent;
+    assert(() {
+      if (debugProfileLayoutsEnabled) {
+        debugTimelineArguments = <String, String>{
+          ...debugTimelineArguments,
+          'dirty count': '${_nodesNeedingLayout.length}',
+          'dirty list': '$_nodesNeedingLayout',
+        };
+      }
+      return true;
+    }());
```

Comment:
Move these inside the `if (!kReleaseMode) {`?

Reply:
good catch. fixed.